File: TOI-Zero_68/A2/006/006.py
from collections import deque

n = int(input())
board = []
for i in range(n):
    board.append(input())

count = 0
visited = [[0] * n for _ in range(n)]
q = deque([(n-1, n-1)])
moves = [(-1, 0), (0, -1)]

while q:
    x, y = q.popleft()
    if visited[x][y]:
        continue
    
    visited[x][y] = 1
    count += 1
    
    for move_x, move_y in moves:
        new_x = x + move_x
        new_y = y + move_y
        
        if (new_x < 0 or new_y < 0 or new_x >= n or new_y >= n or 
            board[new_x][new_y] == 'X'):
            continue
        
        q.append((new_x, new_y))

print(count) 

File: TOI-Zero_68/A2/006/006.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n;
	cin >> n;
	string board[n];
	for(int i=0;i<n;i++){
		cin >> board[i];
	}
	int count=0,visited[n][n];
	memset(visited,0,sizeof(visited));
	queue <pair<int,int> > q;
	q.push({n-1,n-1});
	pair<int,int> move[2]={{-1,0},{0,-1}};
	while(!q.empty()){
		pair<int,int> p=q.front();
		q.pop();
		int x=p.first,y=p.second;
		if(visited[x][y]){
			continue;
		}
		visited[x][y]=1;
		count++;
		for(int i=0;i<2;i++){
			int movex=move[i].first,movey=move[i].second;
			int newx=movex+x,newy=movey+y;
			if(newx<0 || newy<0 || newx>=n || newy>=n || board[newx][newy]=='X'){
				continue;
			}
			q.push({newx,newy});
		}
	}
	cout << count;
}
